datarequest.py

```diff
@@ -1349,10 +1259,377 @@ def mail_datarequest_submitted(ctx, researcher_email, researcher_name, request_i
+        body    = """
+Dear {},
+
+Your data request has been rejected for the following reason(s):
+
+{}
+
+If you wish to object against this rejection, please contact the YOUth data manager.
+
+With kind regards,
+YOUth
+""".format(researcher_name, feedback_for_researcher)
+
+    return mail.send(ctx,
+                     to      = researcher_email,
+                     actor   = user.full_name(ctx),
+                     subject = subject,
+                     body    = body)
+
+
+def mail_datarequest_assignment_submit_assignee(ctx, assignee_email, proposal_title, request_id):
+    return mail.send(ctx,
+                     to      = assignee_email,
+                     actor   = user.full_name(ctx),
+                     subject = "[assignee] YOUth data request {}: assigned".format(request_id),
+                     body    = """
+Dear DMC member,
+
+Data request {} (proposal title: \"{}\") has been assigned to you for review. Please sign in to Yoda to view the data request and submit your review.
+
+The following link will take you directly to the review form: https://portal.yoda.test/datarequest/review/{}.
+
+With kind regards,
+YOUth
+""".format(request_id, proposal_title, request_id))
+
+
+def mail_datarequest_review_submit_researcher(ctx, researcher_email, researcher_name, request_id):
+    return mail.send(ctx,
+                     to      = researcher_email,
+                     actor   = user.full_name(ctx),
+                     subject = "[researcher] YOUth data request {}: reviewed".format(request_id),
+                     body    = """
+Dear {},
+
+Your data request been reviewed by the YOUth data management committee and is awaiting final evaluation by the YOUth Board of Directors.
+
+The following link will take you directly to your data request: https://portal.yoda.test/datarequest/view/{}.
+
+With kind regards,
+YOUth
+""".format(researcher_name, request_id))
+
+
+def mail_datarequest_review_submit_bodmember(ctx, bodmember_email, request_id):
+    return mail.send(ctx,
+                     to      = bodmember_email,
+                     actor   = user.full_name(ctx),
+                     subject = "[bod member] YOUth data request {}: reviewed".format(request_id),
+                     body    = """
+Dear Board of Directors member,
+
+Data request {} has been reviewed by the YOUth data management committee and is awaiting your final evaluation.
+
+Please log into Yoda to evaluate the data request.
+
+The following link will take you directly to the evaluation form: https://portal.yoda.test/datarequest/evaluate/{}.
+
+With kind regards,
+YOUth
+""".format(request_id, request_id))
+
+
+def mail_datarequest_evaluation_submit_researcher(ctx, decision, researcher_email, researcher_name,
+                                                  request_id, datamanager_email = None,
+                                                  feedback_for_researcher = None):
+    if decision == "Approved":
```

Comment:
I think it is cleaner to have this logic outside the mail template. Make separate mail templates and move the logic one level higher.

Reply:
Agreed. Fixed.